menpo/io/input/video.py

```diff
@@ -5,96 +5,106 @@
-            return dict(zip(ffmpeg_exts,
-                        [ImageioFFMPEGImporter] * len(ffmpeg_exts)))
-        except ImportError:
-            return {}
-
-    def build(self):
-        import imageio
-
-        reader = imageio.get_reader(self.filepath, format='ffmpeg', mode='I')
-
-        def imageio_to_menpo(imio_reader, index):
-            pixels = imio_reader.get_data(index)
-            pixels = channels_to_front(pixels)
-
-            if self.normalise:
-                return Image(normalize_pixels_range(pixels), copy=False)
-            else:
-                return Image(pixels, copy=False)
-
-        index_callable = partial(imageio_to_menpo, reader)
-        ll = LazyList.init_from_index_callable(index_callable,
-                                               reader.get_length())
-        ll.fps = reader.get_meta_data()['fps']
-
-        if len(ll) != 0:
-            # TODO: Remove when imageio fixes the ffmpeg importer duration/start
-            # This is a bit grim but the frame->timestamp logic in imageio at
-            # the moment is not very accurate and so we really need to ensure
-            # that the user is returned a list they can actually index into. So
-            # we just remove all the frames that we can't actually index into.
-            # Remove from the front
-            for start in range(len(ll)):
-                if start > 10:  # Arbitrary but probably safe
-                    warnings.warn('Highly inaccurate frame->timestamp mapping '
-                                  'returned by imageio - many frames are being '
-                                  'dropped and thus importing may be very slow.'
-                                  ' Please see the documentation.')
-                try:
-                    ll[start]
-                    break
-                except:
-                    pass
-            else:
-                # If we never broke out then ALL frames raised exceptions
-                ll = LazyList([])
-            # Only take the frames after the initial broken ones
-            ll = ll[start:]
-
-        if len(ll) != 0:
-            n_frames = len(ll) - 1
-            for end in range(n_frames, -1, -1):
-                if end < n_frames - 10:  # Arbitrary but probably safe
-                    warnings.warn('Highly inaccurate frame->timestamp mapping '
-                                  'returned by imageio - many frames are being '
-                                  'dropped and thus importing may be very slow.'
-                                  ' Please see the documentation.')
-                try:
-                    ll[end]
-                    break
-                except:
-                    pass
-            # Only take the frames before the broken ones
-            ll = ll[:end+1]
-
-        return ll
+    import imageio
+
+    reader = imageio.get_reader(str(filepath), format='ffmpeg', mode='I')
+
+    def imageio_to_menpo(imio_reader, index):
```

Comment:
@patricksnape as discussed in person, either pull out of enclosing body (so it isn't a closure) or just used the closed over reader and don't take a parameter for clarity either way. Can be done after @JeanKossaifi adds his video reader though.